tests/phpunit/tests/webfonts-api/wpWebfontsController.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * @group  webfonts
+ * @covers WP_Webfonts_Controller
+ */
+class Tests_Webfonts_API_wpWebfontsController extends WP_UnitTestCase {
+	private $controller;
+	private $webfont_registry_mock;
+	private $provider_registry_mock;
+
+	public static function set_up_before_class() {
+		require_once ABSPATH . WPINC . '/webfonts-api/class-wp-webfonts-registry.php';
+		require_once ABSPATH . WPINC . '/webfonts-api/class-wp-webfonts-provider-registry.php';
+		require_once ABSPATH . WPINC . '/webfonts-api/class-wp-webfonts-controller.php';
+		require_once __DIR__ . '/mocks/class-my-custom-webfonts-provider-mock.php';
+	}
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->webfont_registry_mock  = $this->getMockBuilder( 'WP_Webfonts_Registry' )
+											->disableOriginalConstructor()
+											->getMock();
+		$this->provider_registry_mock = $this->getMockBuilder( 'WP_Webfonts_Provider_Registry' )
+											->getMock();
+		$this->controller             = new WP_Webfonts_Controller(
+			$this->webfont_registry_mock,
+			$this->provider_registry_mock
+		);
+	}
+
+	/**
+	 * @covers WP_Webfonts_Controller::init
+	 *
+	 * @dataProvider data_init
+	 *
+	 * @param string $hook       Expected hook name.
+	 * @param bool   $did_action Whether the action fired or not.
+	 */
+	public function test_init( $hook, $did_action ) {
+		$this->provider_registry_mock
+			->expects( $this->once() )
+			->method( 'init' );
+
+		if ( $did_action ) {
+			do_action( 'wp_enqueue_scripts' );
+		}
+
+		$this->controller->init();
+
+		$this->assertSame(
+			10,
+			has_action( $hook, array( $this->controller, 'generate_and_enqueue_styles' ) )
+		);
+		$this->assertSame(
+			10,
+			has_action( 'admin_init', array( $this->controller, 'generate_and_enqueue_editor_styles' ) )
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_init() {
+		return array(
+			'did_action fired'        => array(
+				'hook'       => 'wp_print_footer_scripts',
+				'did_action' => true,
+			),
+			'did_action did not fire' => array(
+				'hook'       => 'wp_enqueue_scripts',
+				'did_action' => false,
+			),
+		);
+	}
+
+	/**
+	 * By default, the Webfonts API should not request webfonts from
+	 * a remote provider. Test the permissions logic works as expected.
+	 *
+	 * @covers WP_Webfonts_Controller::generate_and_enqueue_styles
+	 *
+	 * @dataProvider data_generate_and_enqueue_editor_styles
+	 *
+	 * @param string $stylestyle_handle Handle for the registered stylesheet.
```

Comment:
```suggestion
	 * @param string $stylesheet_handle Handle for the registered stylesheet.
```
This was missed when updating the parameter name in the function declaration.

Reply:
Whoopsie 🤦‍♀️  Thank you @costdev - fixed in https://github.com/WordPress/wordpress-develop/pull/1736/commits/92bd91195f7896404e666f8a091b7224903157e9.